src/iop/colorin.c

```diff
@@ -1764,6 +1764,89 @@ void reload_defaults(dt_iop_module_t *module)
+
+  char datadir[PATH_MAX] = { 0 };
+  char confdir[PATH_MAX] = { 0 };
+  dt_loc_get_datadir(datadir, sizeof(datadir));
+  dt_loc_get_user_config_dir(confdir, sizeof(confdir));
+  char *system_profile_dir = g_build_filename(datadir, "color", "in", NULL);
+  char *user_profile_dir = g_build_filename(confdir, "color", "in", NULL);
+  char *tooltip_part_profile_dirs = g_strdup_printf(_("darktable loads external ICC profiles from\n%s\nand\n%s"),
+                                                    user_profile_dir, system_profile_dir);
+  g_free(system_profile_dir);
+  g_free(user_profile_dir);
+
+  // In case of embedded ICC profile we will modify tooltip with profile info,
+  // otherwise reset tooltip to generic text.
+  if((color_profile == DT_COLORSPACE_EMBEDDED_ICC) && g)
+  {
+    cmsHPROFILE cmsprofile = cmsOpenProfileFromMem(img->profile, img->profile_size);
+
+    char iccDesc[64]; iccDesc[0] = '\0';
+    cmsGetProfileInfoASCII(cmsprofile, cmsInfoDescription, "en", "US", iccDesc, 64);
+    char iccManuf[64]; iccManuf[0] = '\0';
+    cmsGetProfileInfoASCII(cmsprofile, cmsInfoManufacturer, "en", "US", iccManuf, 64);
+    char iccModel[64]; iccModel[0] = '\0';
+    cmsGetProfileInfoASCII(cmsprofile, cmsInfoModel, "en", "US", iccModel, 64);
+
+    // This is the only profile field in which, although it usually contains a short
+    // copyright text, in theory profile creators can put as long text as they want. So,
+    // we can't take a fast approach of statically allocating memory of some sufficient size
+    // and have to find out the size of the field at run-time and dynamically allocate memory.
+    char* iccCopyr;
+    guint32 bufsize = cmsGetProfileInfoASCII(cmsprofile, cmsInfoCopyright, "en", "US", NULL, 0);
+    if(bufsize)
+    {
+      iccCopyr = malloc(bufsize+1);
+      cmsGetProfileInfoASCII(cmsprofile, cmsInfoCopyright, "en", "US", iccCopyr, bufsize);
+    }
+    else
+    {
+      iccCopyr = "";
+    }
+
+    cmsFloat64Number iccVersion = cmsGetProfileVersion(cmsprofile);
+    char *iccType = "";
+    if(cmsIsMatrixShaper(cmsprofile)) iccType = "Matrix";
+    else if(cmsIsCLUT(cmsprofile, INTENT_PERCEPTUAL, LCMS_USED_AS_INPUT)) iccType = "LUT";
+
+    char *tooltip = g_strdup_printf(_("embedded ICC profile properties:\n\n"
+                                    "name: <b>%s</b>\n"
+                                    "version: <b>%g</b>\n"
+                                    "type: <b>%s</b>\n"
+                                    "manufacturer: <b>%s</b>\n"
+                                    "model: <b>%s</b>\n"
+                                    "copyright: <b>%s</b>\n\n%s"),
+                                    iccDesc,
+                                    iccVersion,
+                                    iccType,
+                                    iccManuf,
+                                    iccModel,
+                                    iccCopyr,
+                                    tooltip_part_profile_dirs);
```

Comment:
> As this pertains to the embedded profile, giving information about system/user profile folders is superfluous here IMHO 

Why?

OK. Images may or may not have an embedded profile. If we work with images without an embedded profile, in this tooltip we will see information about dt's input color profile folders. If we work with images that have an embedded profile, then in your opinion, it is not necessary to show information about folders. Why? Just to save a dozen lines of source code?

> and this code can be made a bit more compact then.

This code definitely does not have such a goal. It should do its job, not be a bit more compact.

> That information is already printed in a tooltip elsewhere for the colorin module, no?

No.

Reply:
> > That information is already printed in a tooltip elsewhere for the colorin module, no?
> 
> No.

Please see below.

Reply:
> Please see below.

Well, it's not _elsewhere in the colorin module_. It's the same tooltip! What you point me to below is the code to set the same tooltip but in gui_init(). BTW, the tooltip set in that code will never be visible because it will be overridden by the setting in reload_defaults().

It's not that I suggest removing that code from gui_init(), but it's essentially redundant. Although in principle this can be done, while leaving a clear comment there why we do not set the tooltip at the beginning, and where is the code that will set the image dependent tooltip.

Reply:
> It's not that I suggest removing that code from gui_init(), but it's essentially redundant.

There's no place for image dependent "initialisations" in `gui_init`; `reload_defaults` always gets called afterwards. This should be cleaned up as people learn from patterns in the code and get confused and continue duplicating stuff "just to be sure". And then bugfixes are applied in the wrong places and everything stays a mess.

Reply:
> And then bugfixes are applied in the wrong places and everything stays a mess.

Very good point! I often try to be minimally invasive to the code in my fixes, but this is a case where it's really better to clean up unnecessary code. Removed with a comment on why it was removed and where the actual tooltip setting is.